sdk/core/azure-core/src/base64.cpp

```diff
@@ -313,10 +313,10 @@ static void Base64WriteIntAsFourBytes(char* destination, int32_t value)
   destination[0] = static_cast<uint8_t>(value & 0xFF);
 }
 
-std::string Base64Encode(const std::vector<uint8_t>& data)
+std::string Base64Encode(uint8_t const* const data, size_t length)
```

Comment:
I assume this split is to avoid a copy?  Comment would be good and we also should put something into our guidelines for this.

Reply:
More specifically, it's to allow callers to avoid a trip through the memory manager.

With the input as a reference to a `std::vector`, the caller *must* create a `std::vector` which allocates a copy on the heap. But there's fundamentally no need for this.

This allows you to create specializations for `Encode` which take the types which you want to encode, currently `std::vector` and `std::string`.

If we were using C++20, I would have changed this to `std::span` which encapsulates the same concept, but we're not, so this is a poor man's approximation.